src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3438 @@
+        gSegments[0x6] = OS_K0_TO_PHYSICAL(play->objectCtx.slots[originalObjectSlot].segment);
+    }
+
+    return ret;
+}
+
+s32 EnAn_ChangeAnim(EnAn* this, PlayState* play, EnAnAnimation animIndex) {
+    s8 originalObjectSlot = this->actor.objectSlot;
+    s8 otherObjectSlot = OBJECT_SLOT_NONE;
+    s32 changeAnim = false;
+    s32 didAnimChange = false;
+
+    if ((animIndex >= ENAN_ANIMOBJ_AN4) && (this->an4ObjectSlot > OBJECT_SLOT_NONE)) {
+        otherObjectSlot = this->an4ObjectSlot;
+    } else if ((animIndex >= ENAN_ANIMOBJ_AN3) && (this->an3ObjectSlot > OBJECT_SLOT_NONE)) {
+        otherObjectSlot = this->an3ObjectSlot;
+    } else if ((animIndex >= ENAN_ANIMOBJ_AN2) && (this->an2ObjectSlot > OBJECT_SLOT_NONE)) {
+        otherObjectSlot = this->an2ObjectSlot;
+    } else if (animIndex < ENAN_ANIMOBJ_AN2) {
+        otherObjectSlot = originalObjectSlot;
+    }
+
+    if (otherObjectSlot <= OBJECT_SLOT_NONE) {
+        return false;
+    }
+
+    switch (animIndex) {
+        case ENAN_ANIM_IDLE:
+        case ENAN_ANIM_IDLE_MORPH:
+            if ((this->animIndex != ENAN_ANIM_IDLE) && (this->animIndex != ENAN_ANIM_IDLE_MORPH)) {
+                changeAnim = true;
+            }
+            break;
+
+        case ENAN_ANIM_WALK:
+        case ENAN_ANIM_WALK2:
+            if ((this->animIndex != ENAN_ANIM_WALK) && (this->animIndex != ENAN_ANIM_WALK2)) {
+                changeAnim = true;
+            }
+            break;
+
+        case ENAN_ANIM_SIT:
+        case ENAN_ANIM_SIT_MORPH:
+            if ((this->animIndex != ENAN_ANIM_SIT) && (this->animIndex != ENAN_ANIM_SIT_MORPH)) {
+                changeAnim = true;
+            }
+            break;
+
+        default:
+            if (animIndex != this->animIndex) {
+                changeAnim = true;
+            }
+            break;
+    }
+
+    if (changeAnim && (otherObjectSlot > OBJECT_SLOT_NONE)) {
+        gSegments[0x6] = OS_K0_TO_PHYSICAL(play->objectCtx.slots[otherObjectSlot].segment);
+        this->animIndex = animIndex;
+        didAnimChange = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, animIndex);
+        this->animPlaySpeed = this->skelAnime.playSpeed;
+        gSegments[0x6] = OS_K0_TO_PHYSICAL(play->objectCtx.slots[originalObjectSlot].segment);
+    }
+
+    return didAnimChange;
+}
+
+void EnAn_UpdateCollider(EnAn* this, PlayState* play) {
+    f32 temp;
+    s32 pad;
+    Vec3f sp24;
+
+    if (this->scheduleResult == ANJU_SCH_WITH_KAFEI) {
+        static Vec3f D_80B58E34 = { 0.0f, 0.0f, 8.0f };
+
+        Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.shape.rot.y, &D_80B58E34, &sp24);
+        this->collider.dim.radius = 20;
+        Math_Vec3f_ToVec3s(&this->collider.dim.pos, &sp24);
+    } else {
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+    }
+
+    temp = this->actor.focus.pos.y - this->actor.world.pos.y;
+    this->collider.dim.height = TRUNCF_BINANG(temp);
```

Comment:
What do you think about calling this variable `height` just based on the following line?